### PR TITLE
Users can add shortcut rules for resources as an option

### PR DESCRIPTION
**Pull request by felixlinker**
This allows users to supply functions that add additional shortcuts for resources. This might come in handy for resources that could otherwise only be accessed by using `api[...]`.
For example have a look at the test case. This rule adds shortcuts for resources like `long-string-with-dashes` in the form of their camel-case counterpart `longStringWithDashes` making this resource accessible via `api.longStringWithDashes`.

### CODE REVIEW

File: src/rest-client.js [hunk 1]
```diff
@@ -136,6 +137,13 @@ function resource(client, parent, name, id, ctx) {
             if (shortcut) {
                 self._shortcuts[resName] = r;
                 self[resName] = r;
+                client._opts.shortcutRules.forEach(rule => {
+                    let customShortcut = rule.call(undefined, resName);
```
Comment (Amareis) [inline src/rest-client.js]: Why not just `rule(resName)`?

**Review by Amareis — CHANGES_REQUESTED** [review]
Add this feature to the readme, please.

File: src/rest-client.js (hunk 1 above)
Comment (felixlinker) [reply src/rest-client.js]: It was intended to lock access to the object. But this should also "break" the this-binding of arrow functions. Good spot, I will replace this line.

**Comment by felixlinker** [comment]
Amended the `README.md` update as well as the usage of `rule(resourceName)` instead of `rule.call(undefined, resourceName)`.

**Comment by Amareis** [comment]
Thanks for your work, I'll upload it to npm in hours.

**Review by Amareis — APPROVED** [review]

**Comment by Amareis** [comment]
Published as 0.5.0.

**Pull request merged**